library/src/main/java/com/tr4android/recyclerviewslideitem/SwipeAdapter.java

```diff
@@ -43,7 +43,7 @@
     public final void onBindViewHolder(RecyclerView.ViewHolder holder, final int position) {
         final RecyclerView.ViewHolder swipeHolder = holder;
         final SwipeItem swipeItem = (SwipeItem) swipeHolder.itemView;
-        SwipeConfiguration configuration = onCreateSwipeConfiguration(swipeItem.getContext(), position);
+        this.configuration = onCreateSwipeConfiguration(swipeItem.getContext(), position);
```

Comment:
@aardouin Do not keep a reference to the `SwipeConfiguration` as a global variable within the `SwipeAdapter`, instead keep it local to the `onBindViewHolder()` method. The following should work:
```java
final SwipeConfiguration configuration = onCreateSwipeConfiguration(swipeItem.getContext(), position);
```
After that you should be able to remove the global declaration of `configuration`. Thanks!